docs/_docs/compute/compute.adoc

```diff
@@ -0,0 +1,63 @@
+= Distributed Computing
+
+Apache Ignite 3 provides an API for distributing computations across cluster nodes in a balanced and fault-tolerant manner. You can submit individual tasks for execution as well as implement the `MapReduce(?)` pattern with automatic task splitting.
+
+Here is how you can execute a remote job by using the `ignite-api` module:
+
+[source, java]
+----
+public interface IgniteCompute {
+// Executes a job represented by the given class on one node from the nodes set.
+<R> CompletableFuture<R> execute(Set<ClusterNode> nodes, Class<? Extends ComputeJob<R>> cls, Object … args);
+
+// Executes a job represented by the given class on one node where the given key is located.
+<R> CompletableFuture<R> executeColocated(String table, K key, Class<? extends ComputeJob<R>> cls, Object … args);
+
+// Executes a job represented by the given class on all nodes from the given nodes set.
+<R> Map<ClusterNode,CompletableFuture<R>> broadcast(Set<ClusterNode> nodes, Class<? extends ComputeJob<R>> cls, Object … args);
+}
+----
+
+NOTE: Currently, this interface supports only basic operations. Broadcasting and operations requiring code deployment are currently not supported.
+
+== Java: Thin
+
+You can use Java thin client to execute compute jobs on the server. Make sure the required classes are deployed to the cluster before executing code.
+
+Here is how you can execute a simple compute job:
+
+[source, java]
+----
+private void example() {
+    IgniteClient client = client();
+    IgniteCompute compute = client.compute();
+    Set<ClusterNode> nodes = new HashSet<>(client.clusterNodes());
+
+    compute.execute(nodes, NodeNameJob.class, "Hello");
+}
+
+private static class NodeNameJob implements ComputeJob<String> {
+    @Override
+    public String execute(JobExecutionContext context, Object... args) {
+        return context.ignite().name() + "_" + args[0];
+    }
+}
+----
+
+NOTE: Unlike Ignite 2, jobs are currently not serialized.
+
+== .NET: Thin
+
+You can use .NET thin client to execute Java compute tasks.  Make sure the required classes are deployed to the cluster before executing code.
+
+Here is how you can execute a simple compute job:
+
+[source, csharp]
+----
+IIgniteClient client = Client;
+ICompute compute = client.Compute;
+IList<IClusterNode> nodes = await Client.GetClusterNodesAsync();
+string res = await compute.ExecuteAsync<string>(nodes, jobClassName: "org.foo.bar.NodeNameJob", "Hello!");
+----
+
+NOTE: Unlike Ignite 2, jobs are currently not serialized.
```

Comment:
```suggestion

```

Reply:
The API only accepts the Java job class name, so there is no possibility of job serialization. I'm not sure if this remark makes sense here.